x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -1331,6 +1349,19 @@ public static String getCreatorRealmName(final Authentication authentication) {
         }
     }
 
+    public static String[] getOwnersRealmNames(Authentication authentication) {
+        if (authentication.isApiKey()) {
+            return new String[] { (String) authentication.getMetadata().get(AuthenticationField.API_KEY_CREATOR_REALM_NAME) };
+        } else {
+            RealmDomain domain = authentication.getSourceRealm().getDomain();
+            if (domain != null) {
+                return domain.realms().stream().map(realmIdentifier -> realmIdentifier.getName()).toArray(String[]::new);
+            } else {
+                return new String[] { authentication.getSourceRealm().getName() };
+            }
+        }
+    }
```

Comment:
Can we add comment to say:
1. How this is different from `getCreatorRealmName`.
2. If the authentication is an API key, only the actual realm is returned since we don't expose owner's domain realms. But otherwise all domain realms are returned.

Reply:
I will comment to that effect, but I think that the real issue is how `manage_own_api_key` privilege works for API Key authentication. We need to discuss this with the team.